web/main.go

```diff
@@ -0,0 +1,344 @@
+	Description string
+}
+
+type Candidate struct {
+	Name                   string
+	Email                  string
+	Party                  string
+	NumberOfTerms          int
+	CandidatureNumber      string
+	Position               string
+	City                   string
+	ImageURL               string
+	TransparencyPercentage int
+	Biography              string
+	SocialLinks            *[]SocialLink
+	Descriptions           *[]CandidateTag
+}
+
+type SocialLink struct {
+	Provider string
+	Link     string
+}
+
+type TeamMember struct {
+	Name        string
+	Title       string
+	ImageURL    string
+	SocialLinks []SocialLink
+}
+
+func newSocialLink(provider string, link string) SocialLink {
+	return SocialLink{
+		Provider: provider,
+		Link:     link,
+	}
+}
+
+func newTeamMember(name string, title string, imageUrl string, socialLinks []SocialLink) TeamMember {
+	return TeamMember{
+		Name:        name,
+		Title:       title,
+		ImageURL:    imageUrl,
+		SocialLinks: socialLinks,
+	}
+}
+
+func newHomeFilters(state string, year string, city string, position string) *HomeFilters {
+	return &HomeFilters{
+		State:    state,
+		Year:     year,
+		City:     city,
+		Position: position,
+	}
+}
+
+func newCandidate() Candidate {
+	return Candidate{
+		Name:                   "Fulado de Tal",
+		Email:                  "fulano@example.com",
+		CandidatureNumber:      "55555",
+		Position:               "Vereador",
+		Party:                  "PSOL",
+		NumberOfTerms:          3,
+		City:                   "Maceió - AL",
+		ImageURL:               "/img/candidata.png",
+		TransparencyPercentage: rand.Intn(100),
+		Biography:              "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci aliquam dignissimos in magnam nihil nostrum optio sint totam unde? Beatae ea illo iusto, laboriosam laudantium libero molestias necessitatibus quos vitae?",
+		SocialLinks: &[]SocialLink{
+			newSocialLink("twitter", "#"),
+			newSocialLink("instagram", "#"),
+			newSocialLink("linkedin", "#"),
+		},
+		Descriptions: &[]CandidateTag{
+			CandidateTag{Tag: "Urbanismo", Description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit."},
+			CandidateTag{Tag: "Veganismo", Description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cum earum iusto nesciunt nobis quaerat quisquam reprehenderit repudiandae temporibus voluptate voluptates. Dolore doloribus expedita, iste laudantium magni nulla pariatur quia totam."},
+		},
+	}
+}
+
+// Define the template registry struct
+type TemplateRegistry struct {
+	templates map[string]*template.Template
+}
+
+// Implement e.Renderer interface
+func (t *TemplateRegistry) Render(w io.Writer, name string, data interface{}, c echo.Context) error {
+	tmpl, ok := t.templates[name]
+	if !ok {
+		err := errors.New("Template not found -> " + name)
+		return err
+	}
+	return tmpl.ExecuteTemplate(w, "layout.html", data)
+}
+
+func getCandidatos(filters *HomeFilters, _offset int) *[]Candidate {
```

Comment:
Aqui acredito que é o mesmo caso que mencionei antes: se você retorna *[]Candidate você tá retornando um ponteiro para um ponteiro. Acho que você quis dizer:
```suggestion
func getCandidatos(filters *HomeFilters, _offset int) []*Candidate {
```

Que seria um slice de ponteiros do tipo Candidate